service-registry/registry-etcd/src/main/java/org/apache/servicecomb/registry/etcd/EtcdRegistration.java

```diff
@@ -0,0 +1,197 @@
+        if (StringUtils.isEmpty(env)) {
+            env = EtcdConst.ETCD_DEFAULT_ENVIRONMENT;
+        }
+        basePath = String.format(EtcdConst.ETCD_DISCOVERY_ROOT, env);
+        etcdInstance = new EtcdInstance();
+        etcdInstance.setInstanceId(registrationId.getInstanceId());
+        etcdInstance.setEnvironment(env);
+        etcdInstance.setApplication(BootStrapProperties.readApplication(environment));
+        etcdInstance.setServiceName(BootStrapProperties.readServiceName(environment));
+        etcdInstance.setAlias(BootStrapProperties.readServiceAlias(environment));
+        etcdInstance.setDescription(BootStrapProperties.readServiceDescription(environment));
+        if (StringUtils.isNotEmpty(dataCenterProperties.getName())) {
+            DataCenterInfo dataCenterInfo = new DataCenterInfo();
+            dataCenterInfo.setName(dataCenterProperties.getName());
+            dataCenterInfo.setRegion(dataCenterProperties.getRegion());
+            dataCenterInfo.setAvailableZone(dataCenterProperties.getAvailableZone());
+            etcdInstance.setDataCenterInfo(dataCenterInfo);
+        }
+        etcdInstance.setProperties(BootStrapProperties.readServiceProperties(environment));
+        etcdInstance.setVersion(BootStrapProperties.readServiceVersion(environment));
+    }
+
+    @Override
+    public void run() {
+        client = Client.builder().endpoints(etcdRegistryProperties.getConnectString())
+                .build();
+        keyPath = basePath + "/" +
+                BootStrapProperties.readApplication(environment) + "/" +
+                BootStrapProperties.readServiceName(environment) + "/" +
+                registrationId.getInstanceId();
+
+        String valueJson = MuteExceptionUtil.builder().withLog("to json, key:{}, value:{}", keyPath, etcdInstance)
+                .executeFunction(JsonUtils::writeValueAsString, etcdInstance);
+        register(ByteSequence.from(keyPath, Charset.defaultCharset()),
+                ByteSequence.from(valueJson, Charset.defaultCharset()));
+    }
+
+    public void register(ByteSequence key, ByteSequence value) {
+
+        Lease leaseClient = client.getLeaseClient();
+        leaseId = MuteExceptionUtil.builder().withLog("get lease id, key:{}, value:{}", keyPath, etcdInstance)
+                .executeCompletableFuture(leaseClient.grant(60)).getID();
+        KV kvClient = client.getKVClient();
+
+        PutOption putOption = PutOption.builder().withLeaseId(leaseId).build();
+        CompletableFuture<PutResponse> putResponse = kvClient.put(key, value, putOption);
+        putResponse.thenRun(() -> {
+            executorService = Executors.newSingleThreadScheduledExecutor();
+            executorService.scheduleAtFixedRate(() -> MuteExceptionUtil.builder().withLog("reRegister, {}, {}", keyPath, etcdInstance)
+                    .executeFunction(leaseClient::keepAliveOnce, leaseId), 0, 5, TimeUnit.SECONDS);
+        });
+    }
+
+    public void unregister() {
```

Comment:
private and maybe you are not using code template in $root/etc.  Please use java chassis project code template to format code.